packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts

```diff
@@ -535,3 +535,12 @@ export interface IDetailsListStyles {
   root: IStyle;
   focusZone: IStyle;
 }
+
+export interface IDetailsGroupRenderProps extends IGroupRenderProps {
+  onRenderFooter?: IRenderFunction<IDetailsGroupDividerProps>;
+  onRenderHeader?: IRenderFunction<IDetailsGroupDividerProps>;
+}
+
+export interface IDetailsGroupDividerProps extends IGroupDividerProps {
```

Comment:
This should probably be:
```ts
detailsListProps: IDetailsListProps; // this.props
columns: IColumn[];
```

This cuts down on work by `DetailsList` to produce these props. Stuff like `itemIndex: -1` can be figured out by your custom component.

Reply:
done